### ISSUE TITLE
First line of dstat output for CPU usage (and others) is always wrong

### ISSUE THREAD

**Issue by jmiserez** [dagwieers/dstat#56]
The first line of dstat output always has wrong values for CPU usage (and others). While this is okay for interactive use, it is not practical for use with the --noupdate option (if I just want 1 line):

With default settings, interactively ("top" says my CPU is really ~88% all the time, this is a database server under load):
```
$ dstat
----total-cpu-usage---- -dsk/total- -net/total- ---paging-- ---system--
usr sys idl wai hiq siq| read  writ| recv  send|  in   out | int   csw 
 57   9  32   1   0   1|  87k 2570k|   0     0 |   0     0 |4035  3331 
 86  13   1   0   0   2|   0  3896k| 342k  640k|   0     0 |6138  5254 
 86  13   0   0   0   1|   0  3468k| 348k  657k|   0     0 |6179  5098 
 85  14   0   0   0   1|   0  3552k| 353k  658k|   0     0 |6240  5164
```

I am running a database server and have a script calls dstat with arguments to only print one line and exit. This is the output:
```
$ dstat -cldnypmgs --tcp -D total --noupdate --nocolor 1 0 
----total-cpu-usage---- ---load-avg--- -dsk/total- -net/total- ---system-- ---procs--- ------memory-usage----- ---paging-- ----swap--- ----tcp-sockets----
usr sys idl wai hiq siq| 1m   5m  15m | read  writ| recv  send| int   csw |run blk new| used  buff  cach  free|  in   out | used  free|lis act syn tim clo
 51   8  39   1   0   1|29.8 30.0 22.9| 105k 2373k|   0     0 |3625  2996 |0.0 0.0 5.3| 395M 93.5M  468M 6498M|   0     0 |   0     0 |  4  92   0   0   0
```
However, the real CPU usage is much higher (top says around 88%). This only shows up after the second line. Here is the output for 3 lines:
```
$ dstat -cldnypmgs --tcp -D total --noupdate --nocolor 1 2 
----total-cpu-usage---- ---load-avg--- -dsk/total- -net/total- ---system-- ---procs--- ------memory-usage----- ---paging-- ----swap--- ----tcp-sockets----
usr sys idl wai hiq siq| 1m   5m  15m | read  writ| recv  send| int   csw |run blk new| used  buff  cach  free|  in   out | used  free|lis act syn tim clo
 53   8  37   1   0   1|28.3 29.6 23.8|  98k 2446k|   0     0 |3776  3118 |0.0   0 4.9| 395M 94.1M  475M 6490M|   0     0 |   0     0 |  4  92   0   0   0
 84  16   0   0   0   1|28.3 29.6 23.8|   0  3664k| 336k  631k|6003  4971 | 35 1.0   0| 395M 94.1M  475M 6490M|   0     0 |   0     0 |  4  92   0   0   0
 85  15   0   0   0   1|28.4 29.6 23.9|   0  4132k| 339k  634k|6038  4912 | 31 1.0   0| 395M 94.1M  476M 6490M|   0     0 |   0     0 |  4  92   0   0   0
```
This also happens when running interactively, the first line always has wrong values for some columns. Also, this happens if I set the window size to a higher value, e.g. one line for 10 seconds:
```
dstat -cldnypmgs --tcp -D total --noupdate --nocolor 10 0
----total-cpu-usage---- ---load-avg--- -dsk/total- -net/total- ---system-- ---procs--- ------memory-usage----- ---paging-- ----swap--- ----tcp-sockets----
usr sys idl wai hiq siq| 1m   5m  15m | read  writ| recv  send| int   csw |run blk new| used  buff  cach  free|  in   out | used  free|lis act syn tim clo
 55   9  35   1   0   1|27.6 28.6 24.2|  93k 2506k|   0     0 |3897  3218 |  0   0 4.7| 395M 94.7M  479M 6486M|   0     0 |   0     0 |  4  92   0   0   0
```

My dstat version is
```
$ dstat --version
Dstat 0.7.0
Written by Dag Wieers <dag@wieers.com>
Homepage at http://dag.wieers.com/home-made/dstat/

Platform posix/linux2
Kernel 3.4.66-55.43.amzn1.x86_64
Python 2.6.8 (unknown, Mar 14 2013, 09:31:22)
[GCC 4.6.2 20111027 (Red Hat 4.6.2-2)]

Terminal type: screen (color support)
Terminal size: 69 lines, 175 columns

Processors: 2
Pagesize: 4096
Clock ticks per secs: 100

internal:
        aio, cpu, cpu24, disk, disk24, disk24old, epoch, fs, int, int24, io, ipc, load, lock, mem, net, page, page24, proc, raw, socket, swap, swapold, sys, tcp, time, udp,
        unix, vm
/usr/share/dstat:
        battery, battery-remain, cpufreq, dbus, disk-util, fan, freespace, gpfs, gpfs-ops, helloworld, innodb-buffer, innodb-io, innodb-ops, lustre, memcache-hits, mysql-io,
        mysql-keys, mysql5-cmds, mysql5-conn, mysql5-io, mysql5-keys, net-packets, nfs3, nfs3-ops, nfsd3, nfsd3-ops, ntp, postfix, power, proc-count, rpc, rpcd, sendmail,
        snooze, thermal, top-bio, top-cpu, top-cputime, top-cputime-avg, top-io, top-latency, top-latency-avg, top-mem, top-oom, utmp, vm-memctl, vmk-hba, vmk-int, vmk-nic,
        vz-cpu, vz-io, vz-ubc, wifi
```

**Comment by dagwieers** [comment]
I am closing this ticket since dstat behaves like all the other tools: the first line immediately calculates the average since system boot.

**Comment by jmiserez** [comment]
Thanks for the the response. I didn't see your comment a year ago, but the reasoning makes sense.

I used dstat to produce plots, which is why I was surprised by this behaviour.

**Comment by gyrovague** [comment]
I ran into this "issue" too, and found this ticket with google. Would it be possible to add something like a "--noinitial" (or --nobootavg or whatever) cmdline option to skip this initial first line? I too found the behaviour surprising, and was also trying to use dstat from a script to automatically measure some CPU and interrupt stats. Right now I've resorted to redirecting the output to a file, and then using head and tail to effectively skip the first line of output after the headers.
